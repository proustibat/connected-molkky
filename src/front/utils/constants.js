export default {
  POSITION: {
    KNOCKED_OVER: 'KNOCKED_OVER',
    UPRIGHT: 'UPRIGHT',
  },

  MESSAGE_TYPE: {
    INFO: 'info',
    WARNING: 'warning',
    ERROR: 'error',
  },
};
