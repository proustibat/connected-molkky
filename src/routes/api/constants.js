export const SKITTLES = {
  AC233FA247EB: 1,
  AC233FA247F0: 2,
  AC233FA247F5: 3,
  AC233FA247F6: 4,
  AC233FA246F1: 5,
  AC233FA247FA: 6,
  AC233FA246FE: 7,
  AC233FA24701: 8,
  AC233FA2470C: 9,
  AC233FA2470D: 10,
  AC233FA246FC: 11,
  AC233FA246FD: 12,
};

export const POSITION_LEVEL = 0.8;
