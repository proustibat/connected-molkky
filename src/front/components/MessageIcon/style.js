import constants from '@utils/constants';
import { colorError, colorInfo, colorWarning } from '@root/gStyles';

const mappers = [{
  predicate: (type) => type === constants.MESSAGE_TYPE.WARNING,
  color: colorWarning,
}, {
  predicate: (type) => type === constants.MESSAGE_TYPE.ERROR,
  color: colorError,
}, {
  predicate: (type) => type === constants.MESSAGE_TYPE.INFO,
  color: colorInfo,
}];

export default {
  container: {
    display: 'flex',
    alignItems: 'center',
  },

  computeStyle: (type) => {
    const { color } = mappers.find((mapper) => mapper.predicate(type));
    return {
      icon: {
        color,
      },
      text: {
        borderBottom: `1px solid ${color}`,
        lineHeight: '1rem',
      },
    };
  },
};
