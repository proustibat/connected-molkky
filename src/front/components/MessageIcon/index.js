import PropTypes from 'prop-types';
import React from 'react';
import constants from '@utils/constants';
import style from './style';

const MessageIcon = ({ type, children }) => (
  <p style={style.container}>
    <i className="small material-icons" style={style.computeStyle(type).icon}>{type}</i>
    <span style={style.computeStyle(type).text}>
      {children}
    </span>
  </p>
);

MessageIcon.propTypes = {
  children: PropTypes.node.isRequired,
  type: PropTypes.oneOf(Object.values(constants.MESSAGE_TYPE)).isRequired,
};

export default MessageIcon;
