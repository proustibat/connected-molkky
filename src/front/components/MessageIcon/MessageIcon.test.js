import React from 'react';
import { shallow } from 'enzyme';
import constants from '@utils/constants';
import MessageIcon from './index';

const givenProps = {
  children: 'Hello World',
};

describe('MessageIcon', () => {
  it('should render the component correctly for warning', () => {
    // Given / When
    const component = shallow(<MessageIcon
      {...givenProps}
      type={constants.MESSAGE_TYPE.WARNING}
    />);

    // Then
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });

  it('should render the component correctly for info', () => {
    // Given / When
    const component = shallow(<MessageIcon
      {...givenProps}
      type={constants.MESSAGE_TYPE.INFO}
    />);

    // Then
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });

  it('should render the component correctly for error', () => {
    // Given / When
    const component = shallow(<MessageIcon
      {...givenProps}
      type={constants.MESSAGE_TYPE.ERROR}
    />);

    // Then
    expect(component).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });
});
