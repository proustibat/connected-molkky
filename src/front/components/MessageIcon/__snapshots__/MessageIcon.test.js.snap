// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageIcon should render the component correctly for error 1`] = `
<p
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
>
  <i
    className="small material-icons"
    style={
      Object {
        "color": "#b71c1c",
      }
    }
  >
    error
  </i>
  <span
    style={
      Object {
        "borderBottom": "1px solid #b71c1c",
        "lineHeight": "1rem",
      }
    }
  >
    Hello World
  </span>
</p>
`;

exports[`MessageIcon should render the component correctly for info 1`] = `
<p
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
>
  <i
    className="small material-icons"
    style={
      Object {
        "color": "#1b5e20",
      }
    }
  >
    info
  </i>
  <span
    style={
      Object {
        "borderBottom": "1px solid #1b5e20",
        "lineHeight": "1rem",
      }
    }
  >
    Hello World
  </span>
</p>
`;

exports[`MessageIcon should render the component correctly for warning 1`] = `
<p
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
>
  <i
    className="small material-icons"
    style={
      Object {
        "color": "#e65100",
      }
    }
  >
    warning
  </i>
  <span
    style={
      Object {
        "borderBottom": "1px solid #e65100",
        "lineHeight": "1rem",
      }
    }
  >
    Hello World
  </span>
</p>
`;
